content/docs/getting-started/hello-world.md

```diff
@@ -1,150 +1,154 @@
 
-Before we begin you should have a very basic understanding of how TCP streams
-work. Having an understanding of Rust’s [standard library
-implementation](https://doc.rust-lang.org/std/net/struct.TcpStream.html) is also
-helpful.
+To check version of Rust on the command line:
+```bash
+rustc --version
+rustc 1.39.0 (4560ea788 2019-11-04)
+```
 
-Let's get started.
+In version `1.39.0`, [Rust introduced async-await], see the [Async Book] for
+in-depth documentation of these new Rust language features. This guide also
+seeks to provide a practical, hands-on introduction for all of the Rust language
+features that are needed for programming with Tokio.
 
-First, generate a new crate.
+Also, most readers will have some experience with writing networking code
+using the [Rust standard library] or another language, though folks who are
+new to network programming should be able to follow along.
 
-```bash
-$ cargo new hello-world
-$ cd hello-world
-```
+# Introducing asynchronous programming
 
-Next, add the necessary dependencies in `Cargo.toml`:
+As our first introduction to [asynchronous programming](../overview) with
+Tokio, we will start with a tiny app that sends "hello, world!" over a
+network connection.
 
-```toml
-[dependencies]
-tokio = { version = "0.2", features = ["full"] }
-```
+The difference between Tokio and a Rust program written using the standard
+library is that Tokio has the flexibility to block program execution
+when the stream is created or when our “hello, world!” message is written
+to the stream.
 
-Tokio requires specifying the requested components using feature flags. This
-allows the user to only include what is needed to run the application, resulting
-in smaller binaries. For getting started, we depend on `full`, which includes
-all components.
+Tokio's [`net`] module provides the same abstraction over networking as
+the corresponding modules in `std` except asynchronously.
 
-Next, add the following to `main.rs`:
+# Communicating through a network connection
 
-```rust
-# #![deny(deprecated)]
-# #![allow(unused_imports)]
+When developing network code, it's helpful to use tools that let us simulate
+each end of the connection. We've chosen two to allow us to receive text
+(*listen* on a socket and *read* data) and send text (*write* data).
 
-use tokio::io;
-use tokio::net::TcpStream;
-use tokio::prelude::*;
+We'll start by sending "hello" over a reliable networking connection ([`TCP`]).
+Before writing code in Rust, let's install and use some network tools to
+manually do what our code will do later in the guide.
 
-#[tokio::main]
-async fn main() {
-    // application comes here
-}
+Install [`socat`](../network-utilities/socat), which is a network utility that
+we'll use to simulate a server. Then type the following command to print
+out everything that is received on port 6142 (a somewhat arbitrary number
+we have chosen for this example):
+
+```bash
+socat TCP-LISTEN:6142 stdout
```

Comment:
Completely unnecessary, but I would suggest changing the command to `socat TCP-LISTEN:6142,fork stdout` so that they don't need to restart `socat` each time their program runs.

Reply:
I like this!